### PR TITLE
Implement new sharecode api endpoint for pilots, fix cache logic

### PR DESCRIPTION
**Pull request by Kuenaimaku**
This implements the new api endpoint for getting pilot data from a share code, as the vault codes have been deprecated.

additionally, this changes how the pilot data is gathered when a pilot is selected from the sync dropdown input after having authenticated - instead of relying on the cloudOwnerID and cloudID, it recreates the key to access aws-storage based on the pilot data that's returned within pilotCache.

note that the api key is currently hard-coded, depending on the results of testing how frequently these endpoints are hit, we may have to create an api key system setting in the future.

### CODE REVIEW

File: src/module/compcon.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { AUDIO_FILE_EXTENSIONS } from "@league-of-foundry-developers/foundry-vtt-types/src/foundry/common/constants.mjs";
```
Comment (mason-bially) [inline src/module/compcon.ts:1]: What's going on here?

**Comment by mason-bially** [comment]
We should probably make an issue for tracking the removal of the depreciated code, there is a lot of stuff going on in there that will no longer need to in the near future.

Besides the strange import I commented on above this looks good.

File: src/module/compcon.ts (hunk 1 above)
Comment (Kuenaimaku) [reply src/module/compcon.ts:1]: That's an interesting auto-include from vscode - i'll remove it in a commit now.

**Comment by Kuenaimaku** [comment]
Agreed, removing that depreciated code and simplifying the sync ui screen in the future might be for the best. Now that we don't have to keep cloudOwnerID and cloudID around, we could probably simplify it using a single string property and attempt to go to the two methods of getting a pilot from comp-con based on regex, instead of the more complex string interpolation.

**Review by Eranziel — APPROVED** [review]

**Pull request merged**